obspy/core/inventory/response.py

```diff
@@ -1663,6 +1663,46 @@ def get_sacpz(self):
         paz = self.get_paz()
         return paz_to_sacpz_string(paz, self.instrument_sensitivity)
 
+    @staticmethod
+    def from_paz(zeros, poles, stage_gain,
+                 stage_gain_frequency=1.0, input_units='M/S',
+                 output_units='VOLTS', normalization_frequency=1.0,
+                 pz_transfer_function_type='LAPLACE (RADIANS/SECOND)',
+                 normalization_factor=1.0):
+        """
+        Convert poles and response lists into a one-stage response.
+
+        Takes in lists of complex poles and zeros and returns a Response with
+        those values defining its only stage.
+        :type zeros: list of complex
+        :param zeros: All zeros of the response to be defined.
+        :type poles: list of complex
+        :param poles: All poles of the response to be defined.
+        :type stage_gain: float
+        :param stage_gain: The gain value of the response [sensitivity]
+
+        Most of the optional parameters defined here are from
+        :class:`~obspy.core.inventory.response.PolesZerosResponseStage`.
+        :returns: new Response instance with given P-Z values
+        """
+
+        pzstage = PolesZerosResponseStage(
```

Comment:
I made the requested changes for these lines of code and just wanted to make sure the comment I set has been seen, since I replied to the request for the change rather than as a main comment in the thread.

I would like to make sure that my test captures the desired behavior for the `recalculate_overall_sensitivity()` line -- for some reason, even if stage_gain_frequency and normalization_frequency are set to the same value, I'm getting a different result for the response's sensitivity compared to the gain for the poles-zeros. This seems like this shouldn't be the case, since the PZ stage is the only one in the response.

To be clear, when I set the stage gain to 2014943641.6, that IS the stage gain for the PZ stage. However, the overall gain looks something like 432115652.548. The frequencies for stage gain and normalization are both kept at 1.0, as is the normalization factor. Is this an issue with the units, or is there a step that I'm missing somewhere?

Reply:
I'll have to check this on monday. The `recalculate_overall_sensitivity()` function just takes the result of evalresp at a given/derived frequency and uses that as the total sensitivity. I think this is correct. But I'll have to look at it in a bit more detail next week to really answer your question.

Reply:
OK, I think we've figured out the issue with this toy test case with the help of someone else here at the lab. I didn't actually make the numbers out of anything but thin air, so the gains of the instrument aren't just the one sensitivity parameter, but also the values associated with the poles and zeros themselves. So I could probably counteract that by making sure the A0 value for the PZ stage would set the response curve to 1 at that point, whereas instead right now the A0 *defaults* to 1, and thus the overall sensitivity is affected by the numbers I apply to the poles and zeros.

Reply:
It looks like the response file is using blockette 58 field 4 to state the gain of stage 1.  I think we want to put that as the normalization so that blockette 53 field 7 has the normalization value (that isn't unity) and then put a sensitivity value for a GS-13 (blockette 58 field 4).